src/plugins/janus_sip.c

```diff
@@ -1801,6 +1801,95 @@ static void janus_sip_sofia_logger(void *stream, char const *fmt, va_list ap) {
 	}
 }
 
+static void janus_sip_sofia_logger_siptrace_single_log_callback(void *stream, char const *fmt, va_list ap) {
+	/* Since the fmt format string in the current Sofia SIP tport_log_msg function implementation is just "%s\n",
+	 * it's more efficient to directly work with the siptrace buffer. */
+	char *buffer = va_arg(ap, char *);
+
+	/* Check if this is a message we need */
+	if(buffer[0] == 's' && buffer[1] == 'e' && buffer[2] == 'n' && buffer[3] == 'd' && buffer[4] == ' ') {
```

Comment:
You're accessing 5 bytes of a buffer that hasn't been validated. I understand that this comes from Sofia, and so it's likely good, but still I'd rather see a `strlen` done on `buffer` before, so that we can check if there are indeed enough bytes for your check.

Reply:
Understood. I will add a `strlen` check on `buffer`.

Reply:
Alternatively, to maintain consistency with other parts of the code, we can use the `strstr` function instead of following the logic of the original logger handler here.